### PR TITLE
Adapt to changes from credentials regarding password length in FIPS mode https://github.com/jenkinsci/credentials-plugin/pull/558

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by jonesbusy — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -126,6 +126,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>credentials</artifactId>
+      <version>1378.v81ef4269d764</version>
```
Comment (pankajy-dev) [inline pom.xml:129]: Why are we providing the version here?

**Review by pankajy-dev — APPROVED** [review]
LGTM

File: pom.xml (hunk 1 above)
Comment (olamy) [reply pom.xml:129]: just need to force version containing this change https://github.com/jenkinsci/jenkins-test-harness/pull/829

Comment (jtnord) [reply pom.xml:129]: not sure I follow, if you want that change, then change the property `jenkins-test-harness.version`?

do you mean that you want a version containing https://github.com/jenkinsci/credentials-plugin/pull/558 instead?

File: src/main/java/hudson/plugins/emailext/MailAccount.java [hunk 2]
```diff
@@ -309,12 +309,19 @@ private void migrateCredentials() {
...
+                newCredentials = new UsernamePasswordCredentialsImpl(
+                        CredentialsScope.GLOBAL,
+                        null,
+                        "Migrated from email-ext username/password",
+                        smtpUsername,
+                        smtpPassword.getPlainText());
+            } catch (Descriptor.FormException e) {
+                // safe to ignore as too short password should happen only in FIPS mode
+                // and migrating from no FIPS to FIPS is not supported, but only fresh start
+                throw new RuntimeException(e);
```
Comment (jtnord) [inline src/main/java/hudson/plugins/emailext/MailAccount.java:323]:
```suggestion
                throw new RuntimeException("Password used for email-ext server configuration could nto be migrated", e);
```

File: pom.xml (hunk 1 above)
Comment (olamy) [reply pom.xml:129]: yup sounds like a typo

**Review by jtnord — APPROVED** [review]

**Comment by olamy** [comment]
@jenkinsci/email-ext-plugin-developers any volunter to merge this to get a release? Thanks

**Review by slide — APPROVED** [review]

**Pull request merged**